src/controller/CHIPDevice.cpp

```diff
@@ -105,6 +105,12 @@ CHIP_ERROR Device::Serialize(SerializedDevice & output)
     memmove(&serializable.mOpsCreds, &mPairing, sizeof(mPairing));
     serializable.mDeviceId   = Encoding::LittleEndian::HostSwap64(mDeviceId);
     serializable.mDevicePort = Encoding::LittleEndian::HostSwap16(mDevicePort);
+#if CHIP_SYSTEM_CONFIG_USE_LWIP
+    serializable.mInterfaceId = 0;
+#else
+    nlSTATIC_ASSERT_PRINT(sizeof(serializable.mInterfaceId) >= sizeof(mInterface), "Size of network interface ID must fit");
```

Comment:
Fwiw, we're using C++14 at this point.  It really is safe to just use `static_assert`....